src/java.base/unix/native/libnio/fs/UnixNativeDispatcher.c

```diff
@@ -593,8 +594,23 @@ static void copy_statx_attributes(JNIEnv* env, struct my_statx* buf, jobject att
     (*env)->SetLongField(env, attrs, attrs_st_atime_sec, (jlong)buf->stx_atime.tv_sec);
     (*env)->SetLongField(env, attrs, attrs_st_mtime_sec, (jlong)buf->stx_mtime.tv_sec);
     (*env)->SetLongField(env, attrs, attrs_st_ctime_sec, (jlong)buf->stx_ctime.tv_sec);
-    (*env)->SetLongField(env, attrs, attrs_st_birthtime_sec, (jlong)buf->stx_btime.tv_sec);
-    (*env)->SetLongField(env, attrs, attrs_st_birthtime_nsec, (jlong)buf->stx_btime.tv_nsec);
+    if ((buf->stx_mask & STATX_BTIME) != 0) {
+        //
+        //  Birth time was filled in so use it
+        //
+        (*env)->SetLongField(env, attrs, attrs_st_birthtime_sec,
+                             (jlong)buf->stx_btime.tv_sec);
+        (*env)->SetLongField(env, attrs, attrs_st_birthtime_nsec,
+                             (jlong)buf->stx_btime.tv_nsec);
+    } else {
+        //
+        //  Birth time was not filled in: fall back to last modification time
+        //
```

Comment:
I guess is okay although I think I'd much prefer to decide this in the Java code. Maybe we can re-examine this again in the future.

Minor nit, can you use the same comment style as the other comments in the code, meaning drop the empty comment lines, only to keep it consistent with the other comments in this function and elsewhere.

Reply:
> I guess is okay although I think I'd much prefer to decide this in the Java code. Maybe we can re-examine this again in the future.

Sure. I think it would be good to get this in now, however, due to backward compatibility.

> Minor nit, can you use the same comment style as the other comments in the code, meaning drop the empty comment lines, only to keep it consistent with the other comments in this function and elsewhere.

Fixed in 347e906.

Do you agree that a CSR is warranted?

Reply:
> Do you agree that a CSR is warranted?

FWIW, the original change didn't have a CSR (only had a release note), so I'm somewhat confused this would need one.

Reply:
It should have as there was a behavioral change: creation time changed from last modified time to epoch when the birth time is not available.

Reply:
Fair enough.

Reply:
> I guess is okay although I think I'd much prefer to decide this in the Java code. Maybe we can re-examine this again in the future.

This is addressed by [JDK-8341282](https://bugs.openjdk.org/browse/JDK-8341282).